freedomofpress/securedrop
# Adds missing Journalist Interface template files to apache2 apparmor profile

## Status

Ready for review 

## Description of Changes

Fixes #5086 .

Added

## Testing

- Check out this branch and use the Molecule upgrade scenario to set up a test env with apparmor:
  ```
  make build-debs
  make upgrade-start
  make upgrade-test-local
  ```
- log in to the app server with `molecule login -s upgrade -h app-staging` and set up an admin account, then retrieve the JI service info from `/var/lib/tor/services/journalist/hostname`
- add a `HidServAuth` line to your local Tor Browser's `torrc` with the JI details
- start Tor Browser and log in to the Ji
  - if you run into problems with  logging in due to mismatched  codes, you may need to generate the code on the app server itself. First install `oathtool` via `apt` and then use the command `oathtool -b --totp "<shared secret here>"`
- [ ] visit the admin section and change the submisison preferences. You should get a confirmation flash message and no application error
- [ ] visit `http://<JI address>/iamapotato`. You should get a 404 page, not an application error.

## Deployment

No special deployment steps required.

## Checklist

### If you made non-trivial code changes:
